clients/src/main/java/org/apache/kafka/clients/consumer/MockConsumer.java

```diff
@@ -55,17 +56,35 @@ public MockConsumer(OffsetResetStrategy offsetResetStrategy) {
     @Override
     public synchronized void subscribe(String... topics) {
         ensureNotClosed();
+        subscribeTopics(topics, false);
+    }
+
+    private void subscribeTopics(String[] topics, boolean isSubscribedViaPatternSubscription) {
+        ensureNotClosed();
         for (String topic : topics)
-            this.subscriptions.subscribe(topic);
+            this.subscriptions.subscribe(topic, isSubscribedViaPatternSubscription);
     }
 
     @Override
     public synchronized void subscribe(TopicPartition... partitions) {
-        ensureNotClosed();
         for (TopicPartition partition : partitions)
             this.subscriptions.subscribe(partition);
     }
 
+    @Override
+    public void subscribe(Pattern pattern) {
+        ensureNotClosed();
+        this.subscriptions.subscribe(pattern);
+        String[] topicsToSubscribe = new String[partitions.size()];
+        int pos = 0;
+        for (String topic: partitions.keySet()) {
+            if (pattern.matcher(topic).matches() &&
+                !subscriptions.subscribedTopics().contains(topic))
+                topicsToSubscribe[pos] = topic;
```

Comment:
@SinghAsDev you are not increasing ``pos``! It should be ``topicsToSubscribe[pos++] = topic``, right?